middleware/file/source/resource/logic.cpp

```diff
@@ -0,0 +1,301 @@
+//! 
+//! Copyright (c) 2015, The casual project
+//!
+//! This software is licensed under the MIT license, https://opensource.org/licenses/MIT
+//!
+
+#include "file/resource/logic.h"
+
+#include "common/communication/instance.h"
+
+#include <expected>
+
+namespace casual::file::resource
+{
+   namespace local
+   {
+      namespace
+      {
+         auto temporary( const common::transaction::ID& transaction, const std::filesystem::path& path)
+         {
+            std::filesystem::path result{ path};
+            result += ".casual.";
+            result += (std::ostringstream{} << transaction).str();
+            return result;
```

Comment:
You could use `common::string::compose( ...)`.
Hmm,  shouldn't it be enough with just the `gtrid`... Or do we need to keep track of the branch for the _temporary_. We only need the transaction part of the filename if the power goes?

Reply:
Do you mean gtrid for just the file naming convention or that the whole resourse manager should keep track of gtrid's only ?

Reply:
"for what" = "for that" :) 
I think we need to keep track of the whole trid. We have to treat different bquals as if the whole trid was different.